wordpress-mobile/WordPress-Android
# [Jetpack Content Migration Flow] Pre flight checks for posts / pages

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/17517

To test:

<details>
<summary>Enable the Jetpack Content Migration Flow</summary>

Enable the following flags in the `build.gradle`

```
        buildConfigField "boolean", "JETPACK_POWERED_BOTTOM_SHEET", "true"
        buildConfigField "boolean", "JETPACK_SHARED_LOGIN", "true"
        buildConfigField "boolean", "JETPACK_LOCAL_USER_FLAGS", "true"
        buildConfigField "boolean", "JETPACK_BLOGGING_REMINDERS_SYNC", "true"
        buildConfigField "boolean", "JETPACK_READER_SAVED_POSTS", "true"
        buildConfigField "boolean", "JETPACK_PROVIDER_SYNC", "true"
        buildConfigField "boolean", "JETPACK_MIGRATION_FLOW", "true"
```

</details>
<details>
<summary>Disable the Jetpack Content Migration Flow</summary>

Enable the following flags in the `build.gradle`

```
        buildConfigField "boolean", "JETPACK_POWERED_BOTTOM_SHEET", "false"
        buildConfigField "boolean", "JETPACK_SHARED_LOGIN", "false"
        buildConfigField "boolean", "JETPACK_LOCAL_USER_FLAGS", "false"
        buildConfigField "boolean", "JETPACK_BLOGGING_REMINDERS_SYNC", "false"
        buildConfigField "boolean", "JETPACK_READER_SAVED_POSTS", "false"
        buildConfigField "boolean", "JETPACK_PROVIDER_SYNC", "false"
        buildConfigField "boolean", "JETPACK_MIGRATION_FLOW", "false"
```

</details>

**Generate some local-only content in the Jetpack App**:
_Uninstall the Jetpack app or log out of Jetpack and delete all data._
1. **Disable the Jetpack Content Migration Flow**
2. Install the Jetpack app flavor from this branch
3. Login to the Jetpack app and select a site
4. Publish a post
5. Make changes to the post and do not publish them (local changes)

**Generate some local-only content in the WordPress App**:

1. **Enable the Jetpack Content Migration Flow**
2. Install the WordPress app flavor from this branch
3. Login to the WordPress app and select a site
4. Publish a post
5. Make changes to the post and do not publish them (local changes)

_You could create more as described in https://github.com/wordpress-mobile/WordPress-Android/pull/17398_

#### Testing steps:

_Uninstall the Jetpack app or log out of Jetpack and delete all data._

1. **Enable the Jetpack Content Migration Flow**
2. Run the Jetpack app from this branch
3. Expect not to see the migration flow
4. Logout
5. Close the app
6. Open the app
7. Expect to be logged in automatically and see the list of your sites
8. Press continue on the next screens
9. Expect to see labels for local content in the post list

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
